src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -3249,7 +3263,10 @@ ss::future<> ntp_archiver::garbage_collect() {
     archival_stm_fence fence = {
```

Comment:
why not something like this? why do we build the fence if it is not gonna be used later?

```cpp
private:
std::option<fence> maybe_fence() const {
  if (_feature_table.local().is_active(features::feature::cloud_storage_metadata_rw_fence))
    return archival_stm_fence{};
  else return {};
}
```

Reply:
I don't see much difference